plugin/plugin.go

```diff
@@ -21,27 +25,77 @@ func New(policy string) validator.Plugin {
 }
 
 func (p *plugin) Validate(ctx context.Context, req *validator.Request) error {
-
-	var droneConfig DroneConfig
-
-	err := yaml.Unmarshal([]byte(req.Config.Data), &droneConfig)
+	f := bytes.NewBufferString(req.Config.Data)
+	resources, err := Parse(f)
 	if err != nil {
 		return err
 	}
+	for _, resource := range resources {
 
-	r := rego.New(
-		rego.Query("deny = data.drone.validation.deny; msg = data.drone.validation.out"),
-		rego.Load([]string{p.policyPath}, nil),
-		rego.Input(droneConfig))
+		r := rego.New(
+			rego.Query("deny = data.drone.validation.deny; msg = data.drone.validation.out"),
+			rego.Load([]string{p.policyPath}, nil),
+			rego.Input(resource))
 
-	rs, err := r.Eval(ctx)
-	if err != nil {
-		return err
+		rs, err := r.Eval(ctx)
+		if err != nil {
+			return err
+		}
+		if rs[0].Bindings["deny"] == true {
+			message := fmt.Sprintf("pipeline %v", rs[0].Bindings["msg"])
+			return errors.New(message)
```

Comment:
**nit:** It might be friendlier to collect errors for all pipelines and return all failures to show a developer everything that's wrong instead of having to potentially step through one failure per-build.

Reply:
I've made a few changes in the latest commit to return a single message to the user with all invalid types.

As it adds some extra complexity to the extension that is specific to this use case (pipeline type validation), future changes in the policy will not be as flexible as they would be with a simpler approach that could require multiple iterations from users.